From 451fd0367e19f6d61aa0dd894d36155bd2921c17 Mon Sep 17 00:00:00 2001
From: Sri Vidya Gunturi <sgunturi@qti.qualcomm.com>
Date: Tue, 8 Apr 2025 10:13:37 +0530
Subject: [PATCH 02/20] UPSTREAM:PROTOCOL: nl80211: Register for Protected EHT
 Action frames

Register for Protected EHT Action frames in the nl80211_action_subscribe_ap
API to receive the below frames -

1. EPCS Priority Access Enable Request Action frame.
2. EPCS Priority Access Enable Response Action frame.
3. EPCS Priority Access Teardown Action frame

Signed-off-by: Sri Vidya Gunturi <sgunturi@qti.qualcomm.com>
---
 src/drivers/driver_nl80211.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index 6aaaf21..2eeda29 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -2831,6 +2831,11 @@ static int nl80211_action_subscribe_ap(struct i802_bss *bss)
 	if (nl80211_register_action_frame(bss, (u8 *) "\x12", 1) < 0)
 		ret = -1;
 #endif /* CONFIG_FST */
+#ifdef CONFIG_IEEE80211BE
+	/* Protected EHT Action frames */
+	if (nl80211_register_action_frame(bss, (u8 *) "\x25", 1) < 0)
+		ret = -1;
+#endif /* CONFIG_IEEE80211BE */
 
 	/* Robust AV */
 	if (nl80211_register_action_frame(bss, (u8 *) "\x13", 1) < 0)
-- 
2.34.1

